packages/trees/frontend/sirius-components-trees/src/trees/Tree.tsx

```diff
@@ -43,18 +43,22 @@ export const Tree = ({ editingContextId, tree, onExpand, selection, setSelection
           const treeItemDomElements = document.querySelectorAll<HTMLElement>('[data-treeitemid]');
           const index = Array.from(treeItemDomElements).indexOf(previousItem);
           const id = dataset.treeitemid;
-          const hasChildren = dataset.haschildren;
-          const expanded = dataset.expanded;
+          const hasChildren = dataset.haschildren === 'true';
+          const isExpanded = dataset.expanded === 'true';
 
           switch (event.key) {
             case 'ArrowLeft':
-              if (hasChildren && expanded === 'true') {
+              if (hasChildren && isExpanded) {
                 onExpand(id, dataset.depth);
+              } else if (index > 0) {
+                treeItemDomElements[index - 1].click();
```

Comment:
While the behavior works for the right arrow, the left arrow should not alway try to go to the previous element.

https://user-images.githubusercontent.com/422587/230095134-84667719-701a-4f7b-8b2c-9e12ba29fdc1.mov

In the following example, it should go to the parent not the previous tree item